nmxact/mgmt/transceiver.go

```diff
@@ -160,6 +203,63 @@ func (t *Transceiver) txRxOmp(txCb TxFn, req *nmp.NmpMsg, mtu int,
 	}
 }
 
+func (t *Transceiver) txRxOmp_async(txCb TxFn, req *nmp.NmpMsg, mtu int,
+	timeout time.Duration, ch chan nmp.NmpRsp, err_c chan error) error {
+
+	seq := req.Hdr.Seq
+	nl, err := t.od.AddNmpListener(seq)
+	if err != nil {
+		err_c <- err
+	}
+
+	var b []byte
+	if t.isTcp {
+		b, err = omp.EncodeOmpTcp(t.txFilterCb, req)
+	} else {
+		b, err = omp.EncodeOmpDgram(t.txFilterCb, req)
+	}
+	if err != nil {
+		err_c <- err
+	}
+
+	log.Debugf("Tx OMP request: %v %s", seq, hex.Dump(b))
+
+	if t.isTcp == false && len(b) > mtu {
+		err_c <- fmt.Errorf("Request too big")
+	}
+	frags := nmxutil.Fragment(b, mtu)
+	for _, frag := range frags {
+		if err := txCb(frag); err != nil {
+			log.Debugf("txCb error %v", err)
+			t.od.RemoveNmpListener(seq)
+			return err
+		}
+	}
+
+	// Now wait for NMP response.
+	go func(seq uint8) {
```

Comment:
You don't need to pass the sequence number into this go routine.  You can just access it directly via the closure.

Reply:
will fix